src/main/java/com/desiato/puresynth/dtos/ErrorLoginResponseDTO.java

```diff
@@ -1,25 +1,18 @@
 package com.desiato.puresynth.dtos;
 
-public class ErrorLoginResponseDTO implements LoginResponseDTO {
-    private final String message;
-    private final boolean success = false;
-
-    public ErrorLoginResponseDTO(String message) {
-        this.message = message;
-    }
-
+public record ErrorLoginResponseDTO(String message, boolean success) implements LoginResponseDTO {
     @Override
     public String getMessage() {
         return message;
     }
 
     @Override
-    public boolean isSuccess() {
-        return success;
+    public String getToken() {
```

Comment:
Adding `token` to the interface is problematic. It doesn't fit the logic, that the error response will have this field. It belongs only to the successful response.

Reply:
ok, it make sense.

Reply:
would you suggest to make the field Optional, or is it better to create two interfaces?
```
public interface LoginResponseDTO {
    String getMessage();
    boolean isSuccess();
}

public interface SuccessfulLoginResponseDTO extends LoginResponseDTO {
    String getToken();
}

Reply:
Please have a look at https://github.com/gdesiato/pure-synth/pull/25#discussion_r1623327991, I don't think you need both interfaces and external exception handler.

Reply:
ok, i think I applied the correct modification. I removed the interfaces, and I eliminated SuccessLoginResponseDTO, leaving only LoginResponseDTO